### PR TITLE
330 remote integration tests for deletes

### LINKED ISSUE #330: Deletes for remote integration test

### PR DESCRIPTION
**Pull request by andreievg**
closes #330 

Branch wouldn't compile. Full changes and branch to use locally when reviewing is form this [PR](https://github.com/openmsupply/open-msupply/pull/335), which also has a [README.md](https://github.com/openmsupply/open-msupply/blob/a83bdbade3259bd5122e846b1e7f9ea6100f8c8a/server/service/src/sync/test/integration/README.MD)

At first i kept previous trait, but realised that it was getting hacky with the extra_data method and adding another trait method for deletes would make it even hackier, so decided to use SyncRecordTest that was used for central data tests, I think it cleaner now, and flexible for extension.

### CODE REVIEW

File: server/service/src/sync/test/integration/remote/mod.rs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+use super::SyncRecordTester;
+
+/// For each test step:
+/// Upsert data to database
+/// Push changes to central server
+/// Reinitilises from cenral server with a fresh database
+/// Check that pulled data matches previously upserted data
+async fn test_remote_sync_record(identifier: &str, tester: &dyn SyncRecordTester) {
+    // util::init_logger(util::LogLevel::Info);
+    random_timeout().await;
```
Comment (clemens-msupply) [inline server/service/src/sync/test/integration/remote/mod.rs:26]: why random_timeout?

**Review by clemens-msupply — APPROVED** [review]
Look good

File: server/service/src/sync/test/integration/remote/mod.rs (hunk 1 above)
Comment (clemens-msupply) [reply server/service/src/sync/test/integration/remote/mod.rs:26]: random_sleep?

Comment (andreievg) [reply server/service/src/sync/test/integration/remote/mod.rs:26]: I changed this once again upstream, now the central data upsert (site creation etc..) use this one:

https://github.com/openmsupply/open-msupply/blob/9c8997df3c00b2227cee8c82c14d7b9cde3df3c9/server/service/src/sync/test/integration/mod.rs#L53-L85

Good point, should have been random_sleep if i kept it

**Pull request merged**